File: solutions/julia/difference-of-squares/1/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
function square_of_sum(n)
	iteration = 0
	sum = 0
	while iteration < n
		iteration += 1

		sum += iteration
	end

	square = sum * sum

	return square
end

"Sum the squares of the first `n` positive integers"
function sum_of_squares(n)
	iteration = 0
	sum = 0
	while iteration < n
		iteration = iteration + 1

		sum += iteration * iteration
	end

	return sum
end

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
function difference(n)
	return square_of_sum(n) - sum_of_squares(n)
end

File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {
	int sum_of_squares (int input) {
		int output = 0;
		for (int i = input; i >= 0; i--) {
			output += i * i;
		}
		return output;
	}

	int square_of_sum (int input) {
		int output = 0;
		for (int i = input; i >= 0; i--) {
			output += i;
		}
		return output * output;
	}

	int difference (int input) {
		if (difference_of_squares::square_of_sum (input) > difference_of_squares::sum_of_squares (input)) {
			return difference_of_squares::square_of_sum (input) - difference_of_squares::sum_of_squares (input);
		} else if (difference_of_squares::square_of_sum (input) < difference_of_squares::sum_of_squares (input)) {
			return difference_of_squares::sum_of_squares (input) - difference_of_squares::square_of_sum (input);
		} else {
			return 0;
		}
	}
}  // namespace difference_of_squares